kiali/kiali
# Setup openId for single cluster

### Describe the change

- Modify scripts to be able to set up openID strategy for single cluster.
- Add a new option to authenticate using groups instead of users directly:
`istio/multicluster/deploy-kiali.sh -ug true`

Note: We might want to move the multicluster scripts modified to the istio directory? As they won't be multicluster exclusive now.

### Steps to test the PR

- `hack/setup-kind-in-ci.sh -a openid`
- Access to Kiali URL 
- Login should work and the user should see namespaces with Kiali user
- Delete clusterRoleBinding kiali-group-viewer
- User should not have permissions

Also test Role/RoleBinding (Needs to have bookinfo installed): 

```
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
 name: kiali-bookinfo
 namespace: bookinfo
rules:
 - apiGroups: [""]
   resources:
   - namespaces
   - pods
   - services
   - replicasets
   verbs:
   - get
   - list
   - watch
```

RoleBinding: 

```
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: kiali-bookinfo
  namespace: bookinfo
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: kiali-bookinfo
subjects:
- kind: Group
  name: oidc:default
  namespace: bookinfo
```

The user should see just the bookinfo namespace: 

![image](https://github.com/user-attachments/assets/ddaa9bbf-a53f-4c25-b192-41d0c88e4173)


### Automation testing

N/A

### Issue reference

Ref. #7472
